File: goblinframework-api/src/main/java/org/goblinframework/api/service/GoblinManagedBean.java
package org.goblinframework.api.service;

import java.lang.annotation.*;

@Documented
@Target(ElementType.TYPE)
@Retention(RetentionPolicy.RUNTIME)
public @interface GoblinManagedBean {

  String type() default "";

  String name() default "";

  boolean register() default true;

}

File: goblinframework-core/src/main/java/org/goblinframework/core/mbean/GoblinManagedBean.kt
package org.goblinframework.core.mbean

@MustBeDocumented
@Target(AnnotationTarget.CLASS, AnnotationTarget.FILE)
@Retention(AnnotationRetention.RUNTIME)
annotation class GoblinManagedBean(val type: String = "",
                                   val name: String = "",
                                   val register: Boolean = true)
